### PR TITLE
Fix ability to change M2O field type and prevent incompatible interfaces in advanced mode

### LINKED ISSUE #10122: Advanced field creation allow using interfaces with incompatible types
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

Advanced field creation allow using any interface with any type, including all incompatible ones.

### To Reproduce

* Navigate to a collection data model setting
* Click on "Create field"
* Pick an interface, e.g. "Map"
* Click "Continue in advanced mode"
* Change the type to anything, e.g. "Boolean"
* Validate
The field will be created with the initial interface selected even if it is incompatible with the new selected type.

### Errors Shown

_No response_

### What version of Directus are you using?

9.1.2

### What version of Node.js are you using?

16

### What database are you using?

PostgreSQL

### What browser are you using?

Firefox

### What operating system are you using?

macOS

### How are you deploying Directus?

locally

### PR DESCRIPTION
**Pull request by oreilles**
Fixes https://github.com/directus/directus/issues/10122
Important parts are:
https://github.com/directus/directus/blob/b8db750cb00153275a90fb0aec7d4877ef8660ad/app/src/modules/settings/routes/data-model/field-detail/store/alterations/standard.ts#L32-L38
https://github.com/directus/directus/blob/b8db750cb00153275a90fb0aec7d4877ef8660ad/app/src/modules/settings/routes/data-model/field-detail/field-detail-simple/field-configuration.vue#L19
https://github.com/directus/directus/blob/b8db750cb00153275a90fb0aec7d4877ef8660ad/app/src/modules/settings/routes/data-model/field-detail/field-detail-simple/field-configuration.vue#L96
Rest of file changes are just updates from all the `interfaces.value.find((interface) => interface.id === something` to `getInterface(something)` ( and similar for `getLayout` and `getDisplays` )

### CODE REVIEW

File: app/src/modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-schema.vue [hunk 1]
```diff
@@ -260,9 +260,7 @@ export default defineComponent({
 
 		const typesWithLabels = computed(() => translate(fieldTypes));
 
-		const typeDisabled = computed(() => {
-			return ['file', 'files', 'o2m', 'm2m', 'm2a', 'm2o', 'translations'].includes(localType.value);
-		});
+		const typeDisabled = computed(() => localType.value !== 'standard');
```
Comment (rijkvanzanten) [inline app/src/modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-schema.vue:263]: This is not the same. You're no longer checking for hash / uuid / json / csv

File: app/src/modules/settings/routes/data-model/field-detail/store/alterations/standard.ts [hunk 2]
```diff
@@ -26,3 +28,11 @@ export function setSpecialForType(updates: StateUpdates) {
 			set(updates, 'field.meta.special', null);
 	}
 }
+
+function updateInterface(updates: StateUpdates, fn: HelperFunctions) {
+	const interfayce = getInterface(fn.getCurrent('field.meta.interface'));
```
Comment (rijkvanzanten) [inline app/src/modules/settings/routes/data-model/field-detail/store/alterations/standard.ts:33]: typo. we'll have to name it something different too as `interface` is a reserved word in JS

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: app/src/modules/settings/routes/data-model/field-detail/store/alterations/standard.ts (hunk 2 above)
Comment (Oreilles) [reply app/src/modules/settings/routes/data-model/field-detail/store/alterations/standard.ts]: Renamed it to `interface_`

File: app/src/modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-schema.vue (hunk 1 above)
Comment (Oreilles) [reply app/src/modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-schema.vue:263]: The `localType` can only be one of those, so the lines should be equivalent ?
https://github.com/directus/directus/blob/5feaac88d7edf68861a050c7c04cd3ab50a6546b/packages/shared/src/constants/fields.ts#L40-L51

Comment (rijkvanzanten) [reply app/src/modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-schema.vue:263]: Ah! Forget I said anything. Got this confused with the regular "TYPES" constant 🙂

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**